### PR TITLE
rust: add pause, resume and detach

### PR DESCRIPTION
**Pull request by hhirtz**
Add functions from the Collection Control API:
https://www.intel.com/content/www/us/en/develop/documentation/vtune-help/top/api-support/instrumentation-and-tracing-technology-apis/instrumentation-tracing-technology-api-reference/collection-control-api.html

### CODE REVIEW

File: rust/ittapi/src/collection_control.rs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+/// - Frame API
+/// - Event API
+/// - User-Defined Synchronization API
+///
+/// See the [Collection Control API] for more details.
+///
+/// # Example
+///
+/// ```
+/// fn main() {
```
Comment (abrown) [inline rust/ittapi/src/collection_control.rs:32]: Looks like our overzealous clippy configuration wants this gone.

File: rust/ittapi/src/lib.rs [hunk 2]
```diff
@@ -13,13 +13,17 @@
...
 mod domain;
 mod event;
 pub mod jit;
 mod string;
 mod task;
 mod util;
 
+pub use collection_control::detach;
+pub use collection_control::pause;
+pub use collection_control::resume;
```
Comment (abrown) [inline rust/ittapi/src/lib.rs:26]: A tiny nit:

```suggestion
pub use collection_control::{detach, pause, resume};
```

**Review by abrown — COMMENTED** [review]
I think this looks good; except for the `fn main` comment below, the CI failures should be mostly gone if you rebase on `master` now that #69 is merged (fixes some CI issues).

**Comment by hhirtz** [comment]
I applied both suggestions to the original commit. clippy doesn't seem to complain anymore :)

**Pull request merged**